storage/rocksdb/ha_rocksdb.cc

```diff
@@ -1325,6 +1311,27 @@ static void rocksdb_set_io_write_timeout(
   RDB_MUTEX_UNLOCK_CHECK(rdb_sysvars_mutex);
 }
 
+int rocksdb_remove_checkpoint(const char *checkpoint_dir_raw) {
+  const auto checkpoint_dir = rdb_normalize_dir(checkpoint_dir_raw);
+  LogPluginErrMsg(INFORMATION_LEVEL, ER_LOG_PRINTF_MSG,
+                  "deleting temporary checkpoint in directory : %s\n",
+                  checkpoint_dir.c_str());
+
+  std::string trash_dir = std::string(rocksdb_datadir) + "/trash";
+  rocksdb::Options op = rocksdb::Options();
+  op.sst_file_manager.reset(NewSstFileManager(
```

Comment:
Not an expert in SST file manager so open questions:
1) can we save reuse the existing manager from `rocksdb_init_internal` as singleton?
2) if no, is it really correct to pass `delete_existing_trash = true` to this one? Wouldn't this be stepping on the toes of the other SST file manager instances?

Reply:
Checked with Rocksdb team. The trash_dir has actually been deprecated. It's the old way of managing trash files. Nowadays Rocksdb team manage them by renaming the sst files. I will pass nullptr to trash dir and set the delete_existing_trash to false since we are not using them.

Reply:
> Checked with Rocksdb team. The trash_dir has actually been deprecated. It's the old way of managing trash files. Nowadays Rocksdb team manage them by renaming the sst files. I will pass nullptr to trash dir and set the delete_existing_trash to false since we are not using them.

I see, thank you. Should we convert the existing `rocksdb_init_internal` usage as well, in this or a follow-up PR? And can we reuse that same manager object?

Reply:
Yes we can convert the existing rocksdb_init_internal usage in a follow-up PR. 

 >And can we reuse that same manager object?

I am open for suggestion. My original thought is that we only have two places use sst_file_manager. Does it worth to maintain a static variable for the usage of these two places?

Reply:
> I am open for suggestion. My original thought is that we only have two places use sst_file_manager. Does it worth to maintain a static variable for the usage of these two places?

I am not sure neither. Clone can run many times, and rotate the checkpoint many times too, so dynamically it could be much more than two places. Also the init options are repeated, and probably should be in sync after the `rocksdb_init_internal` follow-up. A static variable for this would be closely related to `rdb` static var, which looks fine to me.

Reply:
Forgot one more thing, is reusing the same object even safe in the first place?